src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/SpecimenCollectionDateTime.java

```diff
@@ -0,0 +1,100 @@
+    protected static final String KEY = "DTM";
+    private static final String QUALIFIER = "SCO";
+    private static final String KEY_QUALIFIER = KEY + PLUS_SEPARATOR + QUALIFIER;
+    private static final DateTimeFormatter DATE_FORMATTER_CCYYMMDD = DateTimeFormatter.ofPattern("yyyyMMdd");
+    private static final DateTimeFormatter DATE_FORMATTER_CCYYMMDDHHMM = DateTimeFormatter.ofPattern("yyyyMMddHHmm");
+
+    @NonNull
+    private final String collectionDateTime;
+    @NonNull
+    private final DateFormat dateFormat;
+
+    public static SpecimenCollectionDateTime fromString(final String edifactString) {
+        if (!edifactString.startsWith(SpecimenCollectionDateTime.KEY_QUALIFIER)) {
+            throw new IllegalArgumentException(
+                "Can't create " + SpecimenCollectionDateTime.class.getSimpleName() + " from " + edifactString);
+        }
+        final String input = Split.byPlus(edifactString)[1];
+        final String collectionDateTime = Split.byColon(input)[1];
+        final String format = Split.bySegmentTerminator(Split.byColon(input)[2])[0];
+        final SpecimenCollectionDateTimeBuilder collectionDateTimeBuilder = SpecimenCollectionDateTime.builder();
+
+        if (isNotBlank(collectionDateTime) && isNotBlank(format)) {
+            final String formattedFhirDate = getFormattedFhirDate(collectionDateTime, format);
+            collectionDateTimeBuilder
+                .collectionDateTime(formattedFhirDate)
+                .dateFormat(DateFormat.fromCode(format));
+        }
+
+        return collectionDateTimeBuilder.build();
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        return QUALIFIER
+            .concat(COLON_SEPARATOR)
+            .concat(getFormattedEdifactDate(collectionDateTime, dateFormat))
+            .concat(COLON_SEPARATOR)
+            .concat(dateFormat.getCode());
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+        // nothing
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (collectionDateTime.isBlank()) {
+            throw new EdifactValidationException(getKey() + ": Date/time of sample collection is required");
+        }
+    }
+
+    private static String getFormattedFhirDate(final String collectionDateTime, final String dateFormat) {
+        if (dateFormat.equals(DateFormat.CCYYMMDD.getCode())) {
+            return LocalDate.parse(collectionDateTime, DATE_FORMATTER_CCYYMMDD).toString();
+        } else if (dateFormat.equals(DateFormat.CCYYMMDDHHMM.getCode())) {
+            LocalDateTime localDateTime = LocalDateTime.parse(collectionDateTime, DATE_FORMATTER_CCYYMMDDHHMM);
+            return localDateTime.toString() + "+00:00";
+        }
+        throw new UnsupportedOperationException(KEY + ": Date format code " + dateFormat + " is not supported");
```

Comment:
This throws through `fromString` so should we use `IllegalArgumentException`?

Reply:
Sounds like a good idea! 82b6936bfaa9981712252d9ded4dd2258f00fbd8